fudge-ratio/src/main/webapp/code/time.js

```diff
@@ -0,0 +1,53 @@
+class Timer {
+  seconds = 0;
+  minutes = 0;
+  hours = 0;
+
+  interval = null;
+
+  // Concatenates a zero to the beginning of a number less than 10.
+  padd(num) {
+    const padding = num < 10 ? "0" : "";
+    return padding + num;
+  }
+
+  constructor(time) {
+    if (time) {
+      this.hours = parseInt(time.substring(0, 2));
+      this.minutes = parseInt(time.substring(3, 5));
+      this.seconds = parseInt(time.substring(6, 8));
+    }
+  }
+
+  allSeconds() {
+    return this.seconds + this.minutes * 60 + this.hours * 3600;
+  }
+
+  addOneSecond() {
+    this.seconds++;
+    if (this.seconds == 60) {
+      this.minutes++;
+      this.seconds = 0;
+    }
+    if (this.minutes == 60) {
+      this.hours = (this.hours + 1) % 24;
+      this.minutes = 0;
+    }
+  }
+
+  getText() {
+    return this.padd(this.hours) + ":" + this.padd(this.minutes) + ":" + this.padd(this.seconds);
```

Comment:
did you add this method or did we have it before?

Reply:
No, we did not, I add it, tried to add as little as I could, but great eye